### DISCUSSION TITLE
Pass in width/height attributes

### DISCUSSION THREAD

**Discussion by tinyfly** [tailwindlabs/heroicons#1000] [Ideas]
I'd love to be able to add html `width="20"` & `height="20"` attributes to the Vue components and they get passed along to the underlying svg tag. If I make a Vue component out of my own svg image Vue will just pass-through and attributes to the component root tag. But with the heroicons components any thing but a "class" attribute gets stripped out.

I always put hard width and height attributes on all images and svg elements, even if I over ride those with CSS. This does two things:
 1) For images: the browser will make that space in the layout before load so you don't have shifting content as the images come in and, 
2) For svg: if for some reason the css doesn't load at all then the icons don't fill the screen and you still have readable content because they are the size intended.

**Comment by thiemjir** [comment]
Always include width and height attributes as part of the SVG markup. This prevents the SVG from expanding to fill the whole page if the CSS doesn't load (or in scenarios when a user requires custom styles, websites need to be usable even without CSS). Many developers don't know that and just copy the code from Heroicons and modify only the classes.

I hope it's and easy fix that you can do automatically by using the dimensions from the viewbox.

In my point of view it's not a technical debt. It's a bug.
